src/utils/openai.ts

```diff
@@ -97,46 +97,113 @@ const sanitizeMessage = (message: string) => message.trim().replace(/[\n\r]/g, '
 
 const deduplicateMessages = (array: string[]) => Array.from(new Set(array));
 
-const getPrompt = (locale: string, diff: string) => `Write an insightful but concise Git commit message in a complete sentence in present tense for the following diff without prefacing it with anything, the response must be in the language ${locale}:\n${diff}`;
+const getBasePrompt = (locale: string) => `Write an insightful but concise Git commit message in a complete sentence in present tense for the diff that I provide you without prefacing it with anything, the response must be in the language ${locale}`;
+
+const getCommitMessageFormatPrompt = (useConventionalCommits: boolean) => {
+	const commitTitleParts = [];
+
+	if (useConventionalCommits) {
+		commitTitleParts.push('<conventional commits type>(<optional scope of the change>):');
+	}
+
+	commitTitleParts.push('<commit message>');
+
+	return commitTitleParts.join(' ');
+};
+
+const getExtraContextForConventionalCommits = () => {
+	// Based on https://medium.com/neudesic-innovation/conventional-commits-a-better-way-78d6785c2e08
+	const conventionalCommitTypes: Record<string, string> = {
+		/*
+			Commented out feat: fix: change: :refactor because they are too common and
+			will cause the model to generate them too often.
+		*/
+		// feat: 'The commit implements a new feature for the application.',
+		// fix: 'The commit fixes a defect in the application.',
+		// change: 'changes the implementation of an existing feature',
+		// refactor: 'code refactoring',
+		build: 'alters the build system or external dependencies of the product',
+		chore: 'includes a technical or preventative maintenance task',
+		ci: 'continuous integration or continuous delivery scripts or configuration files',
+		deprecate: 'deprecates existing functionality',
+		docs: 'changes to README files and markdown (*.md) files',
+		perf: 'improve the performance of algorithms or general execution',
+		remove: 'removes a feature or dependency',
+		revert: 'reverts one or more commits',
+		security: 'improves security',
+		style: 'updates or reformats the style of the source code',
+		test: 'changes to the suite of automated tests',
+	};
```

Comment:
`deprecate` and `security` are not apart of the Conventional Commits commitlint config or Semantic release so it will be unexpected that when it fails:

- https://github.com/conventional-changelog/commitlint/blob/18fbed7ea86ac0ec9d5449b4979b762ec4305a92/%40commitlint/config-conventional/index.js#L40
- https://github.com/conventional-changelog/conventional-changelog/blob/d0e5d5926c8addba74bc962553dd8bcfba90e228/packages/conventional-changelog-conventionalcommits/writer-opts.js#L182-L193

Reply:
This changelog is interesting. Also the preferred emoji they link to a convensional commit we can use for the Gitmoji PR. I'll remove security and deprecation. I also think they don't add much

Reply:
No worries, I updated it.